src/clio/main/Application.h

```diff
@@ -0,0 +1,229 @@
+#ifndef CLIO_APPLICATION_H
+#define CLIO_APPLICATION_H
+
+#include <memory>
+#include <optional>
+
+#include <boost/asio.hpp>
+#include <boost/asio/ssl.hpp>
+
+#include <clio/main/Config.h>
+
+namespace RPC {
+class Counters;
+class WorkQueue;
+}  // namespace RPC
+
+namespace Backend {
+class BackendInterface;
+}
+
+class DOSGuard;
+class NetworkValidatedLedgers;
+class ReportingETL;
+class ETLLoadBalancer;
+class SubscriptionManager;
+
+class HttpSession;
+class SslHttpSession;
+
+template <class PlainSession, class SslSession>
+class Listener;
+
+namespace Server {
+using HttpServer = Listener<HttpSession, SslHttpSession>;
+}
+
+class Application
+{
+public:
+    virtual ~Application(){};
+
+    virtual Config const&
+    config() const = 0;
+
+    virtual boost::asio::io_context&
+    rpc() const = 0;
+
+    virtual boost::asio::io_context&
+    etl() const = 0;
```

Comment:
These methods should probably be named something else, like `rpcIOCtx()` or something